Compilation fail with headless builds

### Godot version

4.0.dev (f726ba37ec36b2232f0a54ce3576381477ab1aff)

### System information

Arch Linux

### Issue description

Compilation of headless version fails with the following message:
```
platform/linuxbsd/os_linuxbsd.cpp: In member function 'void OS_LinuxBSD::run()':
platform/linuxbsd/os_linuxbsd.cpp:352:17: error: 'DisplayServer' has not been declared
  352 |                 DisplayServer::get_singleton()->process_events(); // get rid of pending events
      |                 ^~~~~~~~~~~~~
```

Command used to compile: `scons -j12 platform=linuxbsd target=release_debug tools=yes x11=no vulkan=no`

Bisected, first bad commit is b396fd4eef36d23fa7325b13a4b8e120f1f000c9 (#57641).

### Steps to reproduce

Compile with `scons -j12 platform=linuxbsd target=release_debug tools=yes x11=no vulkan=no`.

### Minimal reproduction project

_No response_

--- a/platform/linuxbsd/os_linuxbsd.cpp
+++ b/platform/linuxbsd/os_linuxbsd.cpp
@@ -32,6 +32,7 @@
 
 #include "core/io/dir_access.h"
 #include "main/main.h"
+#include "servers/display_server.h"
 
 #ifdef X11_ENABLED
 #include "display_server_x11.h"

